issue #6 Change_of_vars_demo1d.ipynb

pratishthaa opened the pull request:
I haven't used LATEXIFY yet.

karm-patel commented on Change_of_vars_demo1d.ipynb:
```diff
@@ -0,0 +1,250 @@
+{
```
<p>you can give meaningful title in 1st markdown cell.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pyprobml-review/pyprobml-review/pull/7/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

karm-patel replied:
<p>In 2nd iteration you can remove this cell</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pyprobml-review/pyprobml-review/pull/7/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

karm-patel replied:
<pre class="ql-syntax" spellcheck="false">Line #4.    px = 1/(b-a) * jnp.ones(len(xs))
</pre><p>We should follow PEP8 naming convention, which says that variable name should be meaningful.</p><p>here px can be written as prob_x</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pyprobml-review/pyprobml-review/pull/7/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

karm-patel replied:
<pre class="ql-syntax" spellcheck="false">Line #10.    ppy = 1/(2*jnp.sqrt(ys))
</pre><p>same comment for ppy, use some meaningful variable name</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pyprobml-review/pyprobml-review/pull/7/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

karm-patel replied:
<pre class="ql-syntax" spellcheck="false">Line #14.    r=jax.random.PRNGKey(42)
</pre><p>"r" -&gt; "key"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pyprobml-review/pyprobml-review/pull/7/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

karm-patel replied:
<pre class="ql-syntax" spellcheck="false">Line #16.    samples2 = fn(samples)
</pre><p>Here two types of samples are used, so we can give meaningful names, something like</p><p>samples -&gt; uniform_samples</p><p>samples2 -&gt; transformed_samples (or any meaning full name)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pyprobml-review/pyprobml-review/pull/7/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

karm-patel replied:
<pre class="ql-syntax" spellcheck="false">Line #20.    fig, ax = plt.subplots(nrows=1, ncols=3)
</pre><p>These are subplots, so matplotlib default width is too less to show this figure in notebook (as you can observe). I suggest to adjust figsize conditionally, means</p><p>if LATEXIFY:</p><p>	figsize = None #because we do not want to interfere in latexify settings</p><p>else:</p><p>	figsize = (10,4)</p><p> </p><p>I have used this kind of thing in: https://github.com/karm-patel/pyprobml-1/blob/Fig2.6_datasaurus_dozen/notebooks/book1/datasaurus_dozen.ipynb</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pyprobml-review/pyprobml-review/pull/7/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

karm-patel replied:
<pre class="ql-syntax" spellcheck="false">Line #27.    ax[1].set_ylabel("$Pr(y)$")
</pre><p>I think use $p(y)$, idealy you should refer text in textbook to see what notations Dr. murphy is using.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pyprobml-review/pyprobml-review/pull/7/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

karm-patel replied:
<pre class="ql-syntax" spellcheck="false">Line #39.    plt.show()
</pre><ol><li>Here if you observe,the 3rd plot, then yticks is different from the original figure in book1.</li></ol><p>This happens because many a times code in github is not updated as current figure. In this case you need to make some changes, I suggest to read text for this figure, first understand what actually figure is about then may be change some code to make figure same as given in book.</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pyprobml-review/pyprobml-review/pull/7/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

karm-patel commented:
Hi @pratishthaa , Overall code looks good. I have added some comments. In next iterations can you please apply latexify on them? You can show me latexified figures using that latex template and gist.

pratishthaa commented:
Since I used colab, so I added another cell to use latexify and savefig functions of plot_utils.

I have added that changed notebook and latexified image for the same.

pratishthaa commented:
https://gist.github.com/pratishthaa/a7e383586cdad29e85e832589e8bc517

This is the gist for before after

patel-zeel commented on change_of_vars_demo1d/change_of_vars_demo1d.ipynb:
```diff
@@ -0,0 +1,274 @@
+{
```
<pre class="ql-syntax" spellcheck="false">Line #1.    try:
</pre><p>Can you check the new instructions on the README page and modify the code accordingly?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pyprobml-review/pyprobml-review/pull/7/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

patel-zeel replied:
<pre class="ql-syntax" spellcheck="false">Line #11.        latexify(width_scale_factor=2,fig_height=1.5)
</pre><p>We should use width_scale_factor=1 because we need 6 inch wide figure</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pyprobml-review/pyprobml-review/pull/7/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

karm-patel replied:
<pre class="ql-syntax" spellcheck="false">Line #12.    def latexify(
</pre><p>We should not add latexify code in notebook and also figure renders in notebook should not be latexified, Resaon is, as you can see this latexified figure dimensions are lesser than matplotlib default. So when user open notebook in github, he will not able to see it properly.</p><p> </p><p>Since you are doing in colab you are facing problem to run latexify command, right? I can see following solutions</p><ol><li>You can access terminal of google colab with your PC using SSH connection, you need to google how to do that, once you have access of colab's terminal,  then you can do run this command in fork of original repo.</li><li>You can install WSL in your windows PC.  This will enable ubuntu terminal in windows.</li><li>Use VM of ubuntu. (this may take load if you have less RAM)</li><li>Install Ubuntu with windows. (if I were you, I would do this)</li></ol><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pyprobml-review/pyprobml-review/pull/7/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

karm-patel replied:
<pre class="ql-syntax" spellcheck="false">Line #18.    tranisformed_samples = fn(uniform_samples)
</pre><p>correct spelling -&gt; transformed_samples</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pyprobml-review/pyprobml-review/pull/7/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

karm-patel replied:
<pre class="ql-syntax" spellcheck="false">Line #43.      savefig("/content/change_of_vars_demo1d_latexified.pdf") 
</pre><p>Once you setup terminal, fork the main repo and do that all work in that, so you can give actual path need to give to save figure. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pyprobml-review/pyprobml-review/pull/7/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

karm-patel replied:
<pre class="ql-syntax" spellcheck="false">Line #48.    plt.savefig("/content/changeOfVars2.pdf")
</pre><p>You can remove this</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pyprobml-review/pyprobml-review/pull/7/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>
